pair_test.go

```diff
@@ -45,15 +46,23 @@ func makeTestPair(ctx context.Context, t *testing.T) (raft.Transport, raft.Trans
 
 	go func() {
 		<-ctx.Done()
```

Comment:
I think the test would be flaky now. When the test returns, it first cancels the context, allowing this function to close the channels - but there's nothing waiting for this goroutine to finish closing channels before goleak.VerifyNone() is called.

This library is fairly stable (so I don't expect to introduce new goroutine leaks in the future ;)), so maybe the goleak.VerifyNone() calls aren't worth the complexity needed to solve this in the future.

Reply:
I saw the same thing, but didn't want to make a sweeping change. I added a signal channel, created and returned from `makeTestPair()`. It closes after the shutdowns complete, and the tests wait for the channel close in a deferred func.

I left goleak in, because there isn't any downside, and it might prevent a leak in the future. But I can remove if you feel strongly. I'm pretty pleased knowing the connections close.